File: 2657_Find_the_Prefix_Common_Array_of_Two_Arrays/solution.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        n = len(A)
        pca = [0] * n
        for i in range(n):
            c = 0
            for a_index in range(i + 1):
                for b_index in range(i + 1):
                    if A[a_index] == B[b_index]:
                        c += 1
                        break
            pca[i] = c
        return pca

File: 2657_Find_the_Prefix_Common_Array_of_Two_Arrays/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* findThePrefixCommonArray(int* A, int n, int* B, int BSize, int* returnSize) {
    int* pca = (int*)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        int c = 0;
        for (int a = 0; a <= i; a++) for (int b = 0; b <= i; b++) if (A[a] == B[b]) { c++; break; }
        pca[i] = c;
    }
    *returnSize = n;
    return pca;
}
